src/crypto/PublicKey.ts

```diff
@@ -22,9 +22,15 @@ function secp256k1UncompressedCheck(key: secp256k1Uncompressed): void {
   }
 }
 
+const MS_NS_TIMESTAMP_THRESHOLD = new Long(10 ** 9).mul(10 ** 9)
```

Comment:
I think this is too big. I get `1,000,000,000,000,000,000` (19 digits), but when I run `new Long(new Date().valueOf()).mul(1_000_000).toString()` I get a 16 digit number

Reply:
Hmm. Although the test below seems to work. Maybe I'm missing something.

Reply:
Yes, ms timestamps are well below that value and ns timestamps are just above.

Reply:
IOW ms timestamps are just over 10**12 and ns timestamps just over 10**18

Reply:
I think you need to do `Long.fromNumber()` in your expression. I think the constructor truncates the first param to 32 bits.

Reply:
That did the trick